### PR TITLE
OOP School Library: Decorator

### PR DESCRIPTION
**Pull request by raminka13**
## Decorator Pattern
Ruby
#### Interface

- [x] Created a class `Nameable`.
- [x] Implemented a method called `correct_name` that raise a `NotImplementedError`.

#### Turned Person class to Nameable
- [x] Person class inherits from Nameable
- [x] Implemented a method `correct_name`.

#### Prepare base Decorator
- [x] Base `Decorator` inherits from Nameable.
- [x] In the constructor assigned a nameable object from params to an instance variable.
- [x] Implemented the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### CapitalizeDecorator and TrimmerDecorator
- CapitalizeDecorator:
    - [x] Created a class that inherits from the base Decorator class.
    - [x] Implemented a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- TrimmerDecorator:
    - [x] Created a class that inherits from the base Decorator class.
    - [x] Implemented a method `correct_name` that makes sure that the output of `@nameable.correct_name` has a maximum of 10 characters.

### Decorators in Action
Tried the following code to check Decorator over Person:
`person = Person.new(22, 'maximilianus')`
`puts person.correct_name`
`trimmed_person = TrimmerDecorator.new(person)`
`puts trimmed_person.correct_name`
`capitalized_trimmed_person = CapitalizeDecorator.new(trimmed_person)`
`puts capitalized_trimmed_person.correct_name`

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class Nameable
+  def correct_name
+    raise NotImplementedError, 'You must implement the correct method'
+  end
```
Comment (mikethreels) [inline nameable.rb]: - [ ] you might want to be a bit more specific with your error, for example, which class and method the error occurs, see below example:
```"#{self.class} has not implemented method '#{__method__}'"```

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Raminka, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- linters are passing

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: nameable.rb (hunk 1 above)
Comment (raminka13) [reply nameable.rb]: Great suggestion @mikethreels. Will implement it.
And DONE.
Thanks.

**Review by VICTOR-CHUKWU — APPROVED** [review]
Hi @raminka13 , 
# STATUS: APPROVED! 🥇 💯 
done

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
- [ ] All linters passed

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**